ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -166,10 +166,7 @@ private void updateForkChoiceForImportedBlock(
               // a better choice than the block itself.  So the first block we receive that is a
               // child of our current chain head, must be the new chain head. If we'd had any other
               // child of the current chain head we'd have already selected it as head.
-              if (recentChainData
-                  .getHeadBlock()
-                  .map(currentHead -> currentHead.getRoot().equals(block.getParent_root()))
-                  .orElse(false)) {
+              if (recentChainData.isChildOfChainHead(block)) {
```

Comment:
Should we not be checking is the new block is the **ancestor** of chain head instead?

Reply:
We can't be importing an ancestor of the chain head because we'd have to have imported the child first and would then have already updated the chain head to be that child.